src/sorting.jl

```diff
@@ -578,10 +578,43 @@ end
     end
 end
 
+function extraneous_block(vals :: AbstractArray, dims):: Bool
+    other_linear_index = ((gridDim().z  ÷ blockDim().z) * (blockIdx().y - 1)) + blockIdx().z
+    return other_linear_index > length(vals) ÷ size(vals)[dims]
+end
+
+function extraneous_block(vals, dims) :: Bool
+    return extraneous_block(vals[1], dims)
+end
+
+# methods are defined for Val{1} because using view has 2x speed penalty for 1D arrays
```

Comment:
We could probably speed up `view` by returning a `CuDeviceArray` when possible, just like we do with `CuArray`

Reply:
Are you suggesting a change here or is this an idea for a separate change in the repo?

Reply:
A separate change.